File: container-with-most-water/container-with-most-water.go
func maxArea(height []int) int {
    var m int
    var z, t int
    var l bool
    var j = len(height) - 1

    for i := 0; i< j; {
        l = true
        z = height[i]
        if z > height[j] {
            l = false
            z = height[j]
        }
        t = (j - i) * z
        if m < t {
            m = t
        }
        if l {
            i++
        } else {
            j--
        }
    }

    return m
}

File: container-with-most-water/container-with-most-water.c
int maxArea(int* height, int heightSize){
    int m = 0;
    int z = 0, t = 0;
    bool l = true;

    for (int i = 0, j = heightSize - 1; i < j;) {
        z = height[i];
        l = true;
        if (height[i] > height[j]) {
            z = height[j];
            l = false;
        }
        t = (j -i) * z;
        if (t > m) {
            m = t;
        }
        if (l) {
            i++;
        } else {
            j--;
        }
    }

    return m;
}
